modules/twinklexfd.js

```diff
@@ -1493,34 +1474,21 @@ Twinkle.xfd.callbacks = {
 Twinkle.xfd.callback.evaluate = function(e) {
 	var form = e.target;
 
-	var type = form.category.value;
-	var usertalk = form.notify.checked;
-	var reason = form.xfdreason.value;
-	var delsort_cats = $(form.delsort).val(); // afd
-	var xfdcat = form.xfdcat && form.xfdcat.value; // afd, cfd, cfds, tfd
-	var xfdtarget = form.xfdtarget && form.xfdtarget.value; // cfd, cfds, tfd
-	var xfdtarget2 = form.xfdtarget2 && form.xfdtarget2.value; // cfd
-	var noinclude = form.noinclude && form.noinclude.checked; // afd, mfd, tfd
-	var tfdtype = form.templatetype && form.templatetype.value; // tfd
-	var notifyuserspace = form.notifyuserspace && form.notifyuserspace.checked; // mfd
-	var relatedpage = form.relatedpage && form.relatedpage.checked; // rfd
-	var newname = form.newname && form.newname.value; // rm
-	var rmtr = form.rmtr && form.rmtr.checked; // rm
-
+	var input = Morebits.quickForm.getInputData(form);
```

Comment:
@siddharthvp This comment should make more sense now that it's been rebased.  Basically:

1. L1697 becomes `var params = Morebits.quickForm.getInputData(form);`, we can remove the rest of the `$.extend`s
1. The various `if (input.xfdtarget) {`-like stuff in the switch becomes `if (params.xfdtarget) {`

I'm not 100% on if there are downsides.  It'd be places where one venue expects certain parameters to be missing or something.  `getDiscussionWikitext` would be the likely place, but worth checking beforehand.

Reply:
Yeah, this would definitely be great - only with this the full power of `getInputData` is leveraged. 

This and a lot more has been added in a new commit. The cleanup is rather extensive, but if all that was put in a separate PR, it would have created the doublework of testing everything all over again. I'm not confident I've verified everything though; the UI is definitely working, testing the template outputs on the other hand, is quite troublesome given testwiki's lack of templates. 

God that was a truckload of shitty code.